op-tests/test-bls-ops.txt

```diff
@@ -22,6 +22,117 @@ coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789
 coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 -1 => FAIL
 coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 0x00001234 => FAIL
 
+; modpow
+modpow => FAIL
+modpow 1 => FAIL
+modpow 1 2 => FAIL
+modpow 1 2 3 4 => FAIL
+modpow (1 2) 2 3 => FAIL
+modpow 1 (2 2) 3 => FAIL
+modpow 1 2 (2 3) => FAIL
+
+; raised to negative
+modpow 999 -1 10 => FAIL
+modpow 2 -1 10 => FAIL
+
+; mod of 0
+modpow 2 2 0 => FAIL
+modpow 2 3 0 => FAIL
+
+modpow 2 0 100 => 1 | 11033
+modpow 2 1 100 => 2 | 11036
+modpow 2 2 100 => 4 | 11036
+modpow 2 3 100 => 8 | 11036
+
+modpow 2 0 -10 => -9 | 11033
+modpow 2 1 -10 => -8 | 11036
+modpow 2 2 -10 => -6 | 11036
+modpow 2 3 -10 => -2 | 11036
+
+modpow 2 0 10 => 1 | 11033
+modpow 999 0 10 => 1 | 11048
+
+modpow 12345 6789 4444 => 2105 | 11094
+modpow -12345 6789 4444 => 2339 | 11094
+modpow -12345 6789 -4444 => -2105 | 11094
+modpow 12345 6789 -4444 => -2339 | 11094
+
+modpow 12345 6789 44444444444 => 13456191581 | 11292
+
+modpow 0 6789 4444 => 0 | 11044
+modpow 2 64 32 => 0 | 11026
+modpow 2 64 31 => 16 | 11036
+
+; mod
+% 0 3 => 0 | 992
+% 1 3 => 1 | 1006
+% 2 3 => 2 | 1006
+% 3 3 => 0 | 996
+% 0 -3 => 0 | 992
+% 1 -3 => -2 | 1006
+% 2 -3 => -1 | 1006
+% 3 -3 => 0 | 996
+
+% 10 3 => 1 | 1006
+% -10 3 => 2 | 1006
+% -10 -3 => -1 | 1006
+% 10 -3 => -2 | 1006
+
+% 1 3 => 1 | 1006
+% -1 3 => 2 | 1006
+% -1 -3 => -1 | 1006
+% 1 -3 => -2 | 1006
+
+% 1 1 => 0 | 996
+% -1 1 => 0 | 996
+% -1 -1 => 0 | 996
+% 1 -1 => 0 | 996
+
+% 1 10 => 1 | 1006
+% 1 -10 => -9 | 1006
+% -1 -10 => -1 | 1006
+% -1 10 => 9 | 1006
+% 1 -1000000000000 => -999999999999 | 1076
+% -1 1000000000000 => 999999999999 | 1076
+
+% 80001 73 => 66 | 1014
+% -80001 73 => 7 | 1014
+% -80001 -73 => -66 | 1014
+% 80001 -73 => -7 | 1014
+
+% 80000 -10 => 0 | 1004
+% 80000 10 => 0 | 1004
+% -80000 10 => 0  | 1004
+
+% 0x00000000000000000a 0x000000000000000005 => 0 | 1060
+% 0x000000000000000000000000000000000000000000000000000000000000013881 0x000000000000000000000000000000000000000000000000000049 => 66 | 1238
+
+% -10 -7 => -3 | 1006
+% -10 7 => 4 | 1006
+% 10 -7 => -4 | 1006
+% 10 7 => 3 | 1006
+% -10 -70 => -10 | 1006
+% -10 70 => 60 | 1006
+% 10 -70 => -60 | 1006
+% 10 70 => 10 | 1006
+% -100 -7 => -2 | 1006
+% -100 7 => 5 | 1006
+% 100 -7 => -5 | 1006
+% 100 7 => 2 | 1006
+% -100 -70 => -30 | 1006
+% -100 70 => 40 | 1006
+% 100 -70 => -40 | 1006
+% 100 70 => 30 | 1006
+
+; mod by zero
+% 3 0 => FAIL
+; invalid arguments
+% 3 => FAIL
+% => FAIL
+% 1 2 3 => FAIL
+% (1 2 3 ) 2 => FAIL
+% 2 (1 2 3 ) => FAIL
```

Comment:
So these match the `divmod` values?

Reply:
yes. I took many of the same test cases from `divmod`